### DISCUSSION TITLE
Reliable 240AC Current sensor

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16333] [Q&A] [answered]
I'd like to monitor the output of my solar system.
The strings are all connected to their own breakers in a breaker panel with a single output going to the city meter.

It has four 2600KW strings and a final output of 10400KW.
I'd like to monitor each string and the final output.

However, I need a device that will be pretty accurate so I can compare the output to the PLC data being sent now.

**Comment by barbudor** [comment, marked as answer]
Are you sure of 10400 KW? 40000 A? 
That's 10 MW so pretty the capacity of a small power plant to feed a small city

**Reply by barbudor** [reply]
If you go to more reasonable currents, Shelly 3EM has 3 channels of 120A each so 93kW total

**Reply by dac2020** [reply]
LOL, I forgot the decimal and forgot to mention that it's all 240VAC.
It's 42 panels, four strings of 2600 kilowatt each. The system is total 10.4KW. 

Each string is connected to one 30amp breaker but it would not put that much out of course.

**Reply by barbudor** [reply]
You can also look at https://tasmota.github.io/docs/PZEM-0XX/

**Reply by dac2020** [reply]
Thanks. I looked at that just now. Before doing a board, I'd like to find something pre-built with case if I can.
I've not had much luck with boards :). 

I'll take a look at the Shelly's if that's what you feel are most reliable. 

Thank you.

**Reply by pkkrusty** [reply]
I've had good success with PZEM-004T. For only $6 each, can't go wrong.

**Reply by dac2020** [reply]
This one?

https://www.amazon.com/PZEM-004T-Current-Voltage-Multimeter-Transformer/dp/B074QGWMKH

I need a clamp I can screw on rather than having to disconnect the power lines. I also need the clamps to be very small as there is limited space for 5 of them.
And of course, I'd like them to be wireless.

**Reply by pkkrusty** [reply]
US $25.97  2％ Off | 5Pcs AC 80-260V 0-100A Digital Multifunction Meter Watt Power Volt Current Test Module PZEM-004T For Arduino TTL COM2\COM3\COM4
https://a.aliexpress.com/_mqQFc9q

**Reply by dac2020** [reply]
I'll keep it in mind but I'm really looking for something that can work with wifi and mqtt. 

Thank you.

**Reply by pkkrusty** [reply]
Ah, it hooks into a ESP8266 module via Tx/Rx. Sounds like you're looking for something more all-in-one.

**Reply by dac2020** [reply]
Since I need to monitor five loads, price is an issue but also complexity. 
The panel is outside and running wires is quite difficult now. 
Wireless devices I could put inside the panel would be the best way possible in this case.
Now, if that means boards like you showed me and one ESPxxx module that sends all the data, that could work. 
I would just have to make sure it's all waterproof.

**Reply by barbudor** [reply]
I think the Shelly EM is what will best suit for you : ready device, no hacking, compact and din rail 1 U

PZEM is bigger so you will need more room.
There are some constraint I think to connect multiple to 1 tasmota. I think max is 3 but may depend on the PZEM model.
You would end up with 2 ESP board like mini D1/nodeMCU + 5V PSU.
I found nice DIN Rail PSU on Ali. I'm using one for powering a Pi Z2 inside a 1 unit DIN box.

**Reply by dac2020** [reply]
Yes, it looks like the Shelly might do a good job. Just depends on the current transformers used perhaps.
I've no idea if a loose fit is better than a tight one. I don't know if there's a way to monitor current usage in-line or only with a current transformer? Lots to learn still.

**Reply by barbudor** [reply]
transformer is the safer as there is perfect isolation between the main crcuit and the measurment circuit

I don't think I've seen in-line (ie measuring volatge across a resistor) for currents higher than 16-20A
generally they are switches with power measurments such as
- Sonof POW R3 25A https://sonoff.tech/product/diy-smart-switch/powr3/ https://templates.blakadder.com/sonoff_POWR3.html
- Sonoff POW Origin 16A https://sonoff.tech/product/diy-smart-switch/pow-origin/ https://templates.blakadder.com/sonoff_POWR316.html
- Sonoff POW Elite 20A https://sonoff.tech/product/diy-smart-switch/pow-elite/ https://templates.blakadder.com/sonoff_POWR320D.html
- Sonoff SPM is quite interresting as it is a main controller unit with ESP32 and slave units with 8 relay/measurment each and you can have up to 32 slave so 128 relay/measurment in total with 20A per channel https://sonoff.tech/product/spm-main-spm-4relay/

**Reply by dac2020** [reply]
The Shelly PM seem to have the CT built in. These Sonoff look interesting too. Thank you!
